remotespark/sparkkernelbase.py

```diff
@@ -9,35 +9,40 @@
 
 
 class SparkKernelBase(IPythonKernel):
-    # Required by Jupyter - Override
-    implementation = None
-    implementation_version = None
-    language = None
-    language_version = None
-    language_info = None
-    banner = None
-
-    # Override
-    kernel_conf_name = None
-    session_language = None
-    client_name = None
-
-    def __init__(self, **kwargs):
+    def __init__(self, implementation, implementation_version, language, language_version, language_info,
+                 kernel_conf_name, session_language, client_name, **kwargs):
+        # Required by Jupyter - Override
+        self.implementation = implementation
+        self.implementation_version = implementation_version
+        self.language = language
+        self.language_version = language_version
+        self.language_info = language_info
+
+        # Override
+        self.kernel_conf_name = kernel_conf_name
+        self.session_language = session_language
+        self.client_name = client_name
+
         super(SparkKernelBase, self).__init__(**kwargs)
+
         self.logger = Log(self.client_name)
-        self.already_ran_once = False
+        self.session_started = False
         self._fatal_error = None
 
         # Disable warnings for test env in HDI
         requests.packages.urllib3.disable_warnings()
 
+        if "testing" not in kwargs.keys():
```

Comment:
Try `if not kwargs.get("testing", False)`.  This will work even if I pass in `testing=False` as a keyword argument